### PR TITLE
Merge feature branch

### PR DESCRIPTION
**Pull request by gcosne**
This branch aim to refactor the different versions in a same repo

### CODE REVIEW

File: utils.py [hunk 1]
```diff
@@ -236,6 +340,97 @@ def load_vgg16(model_dir):
...
+            x = adjust_input_image_size_for_proper_feature_alignment(x, output_stride=8)
+
+        x = self.resnet34_8s(x)
+
+        x = nn.functional.upsample_bilinear(input=x, size=input_spatial_dim)
+
+        return x
+
+def load_segmentation_model(ckpt_path):
+    model = Resnet34_8s(num_classes=19).to('cuda')
```
Comment (gcosne) [inline utils.py]: why this resnet in particular ( refer to github issue ), num_classes is fixed

File: trainer.py [hunk 2]
```diff
@@ -182,7 +221,29 @@ def compute_vgg_loss(self, vgg, img, target):
...
+        target_with_mask = torch.mul(1-mask1_tensor,target) +mask1_tensor*19
+        mask2 = torch.nn.functional.interpolate(mask, size=(self.newsize,self.newsize))
+        mask_tensor = torch.tensor(mask2,dtype=torch.float).cuda()
+        output_with_mask = (torch.mul(1-mask_tensor,output))  
+        # cat the mask as to the logits (loss=0 over the masked region)
+        output_with_mask_cat = torch.cat((output_with_mask.squeeze(),mask_tensor.squeeze().unsqueeze(0)))
+        loss = nn.CrossEntropyLoss()(output_with_mask_cat.unsqueeze(0),target_with_mask.squeeze().unsqueeze(0))
+        return(loss)
     def sample(self, x_a, x_b):
         self.eval()
```
Comment (gcosne) [inline trainer.py]: comment the function: loading the images for inferences + the segmentation mask before and after translation
